recipes/bx/all/conanfile.py

```diff
@@ -1,81 +1,239 @@
-            raise ConanInvalidConfiguration("%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
+        if self.settings.os == "Windows":
+            self.lib_ext = "*.lib"
+            self.package_lib_prefix = ""
+        elif self.settings.os in ["Linux", "FreeBSD"]:
+            self.lib_ext = "*.a"
+            self.package_lib_prefix = "lib"
+        elif self.settings.os == "Macos":
+            self.lib_ext = "*.a"
+            self.package_lib_prefix = "lib"
+        
+        self.genie_extra = ""
+        self.projs = ["bx"]
+        if self.options.tools:
+            self.projs.extend(["bin2c", "lemon"])
+        if is_msvc(self) and not is_msvc_static_runtime(self):
+            self.genie_extra += " --with-dynamic-runtime"
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def validate(self):
+        if not self.options.get_safe("fPIC", True):
+            raise ConanInvalidConfiguration("This package does not support builds without fPIC.")
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, 14)
+        check_min_vs(self, 191)
+        if not is_msvc(self):
+            try:
+                minimum_required_compiler_version = self._compiler_required[str(self.settings.compiler)]
+                if Version(self.settings.compiler.version) < minimum_required_compiler_version:
+                    raise ConanInvalidConfiguration("This package requires C++14 support. The current compiler does not support it.")
+            except KeyError:
+                self.output.warn("This recipe has no checking for the current compiler. Please consider adding it.")
+
+    def build_requirements(self):
+        self.tool_requires("genie/1170")
+        if not is_msvc(self) and self._settings_build.os == "Windows":
+            self.win_bash = True
+            if not self.conf.get("tools.microsoft.bash:path", check_type=str):
+                self.tool_requires("msys2/cci.latest")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = glob.glob('bx-*/')[0]
-        os.rename(extracted_dir, self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True,
+                    destination=os.path.join(self.source_folder, self._bx_folder))
 
-    def _configure_cmake(self):
-        cmake = CMake(self)
-        cmake.configure(build_folder=self._build_subfolder)
-        return cmake
+    def generate(self):
+        vbe = VirtualBuildEnv(self)
+        vbe.generate()
+        if is_msvc(self):
+            tc = VCVars(self)
+            tc.generate()
+        else:
+            tc = AutotoolsToolchain(self)
+            tc.generate()
 
     def build(self):
-        cmake = self._configure_cmake()
-        cmake.build()
+        # Save bx path for convencience
+        self.bx_path = os.path.join(self.source_folder, self._bx_folder)
```

Comment:
This could be a property

```py
    @property
    def bx_path(self):
        return os.path.join(self.source_folder, self._bx_folder)
```

This would address the linters complaints

```suggestion
```